src/main/java/com/google/api/generator/gapic/composer/Composer.java

```diff
@@ -186,6 +188,40 @@ public static List<GapicClass> generateTestClasses(GapicContext context) {
     return clazzes;
   }
 
+  @VisibleForTesting
+  static List<GapicClass> composeSamples(List<GapicClass> clazzes, String protoPackage) {
+    //  parse protoPackage for apiVersion and apiShortName
+    String[] pakkage = protoPackage.split("\\.");
+    String apiVersion;
+    String apiShortName;
+    //  e.g. v1, v2, v1beta1
+    if (pakkage[pakkage.length - 1].matches("v[0-9].*")) {
+      apiVersion = pakkage[pakkage.length - 1];
+      apiShortName = pakkage[pakkage.length - 2];
+    } else {
+      apiVersion = "";
+      apiShortName = pakkage[pakkage.length - 1];
+    }
+    //  Include license header, apiShortName, and apiVersion
+    return clazzes.stream()
+        .map(
+            gapicClass -> {
+              List<Sample> samples =
+                  gapicClass.samples().stream()
+                      .map(sample -> updateSample(sample, apiShortName, apiVersion))
+                      .collect(Collectors.toList());
+              return gapicClass.withSamples(samples);
+            })
+        .collect(Collectors.toList());
+  }
+
+  private static Sample updateSample(Sample sample, String apiShortName, String apiVersion) {
```

Comment:
The method could be renamed to something like `addRegionTagAndHeaderToSample` to make it more specific, `updateSample` is too generic and does not indicate what exactly it's doing.

Reply:
done